src/main/jbake/content/blog/2017/01/13/events-aggregation-extension.adoc

```diff
@@ -0,0 +1,189 @@
+= Extending Complex Event Processing in Hawkular Alerting
+Lucas Ponce
+2017-01-13
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, alerts, complex, events, processing, stream, polling
+
+== Stream Processing versus Polling Processing
+
+Hawkular Alerting uses different techniques to detect behaviours by defining rules.
+
+The main implementation is based in a Stream Processing design.
+Hawkular Alerting analyzes streams of data and events ordered in a timeline searching for matches against conditions.
+This technique is also known as Event Processing (or Complex Event Processing) specially when the processing of incoming data might need multiple conditions.
+
+ifndef::env-github[]
+image::/img/blog/2017/stream-processing.png[Stream Processing]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/stream-processing.png[Stream Processing]
+endif::[]
+
+This method is particularity good to identify meaningful happenings on complex scenarios and respond to them as quickly as possible.
+
+Hawkular Alerting also supports a Polling technique based on _Alerters_ that allow us to define queries that are processed
+against a backend in a periodic manner.
+
+ifndef::env-github[]
+image::/img/blog/2017/polling-processing.png[Polling Processing]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/polling-processing.png[Polling Processing]
+endif::[]
+
+This Polling technique is also good in non real time scenarios, for example, to compare historical values in
+large range time intervals using statistical operators like averages, medians or percentiles.
+
+== Sliding Windows on Complex Event Processing
+
+An interesting aspect in the logic of detection behaviours is when we can combine characteristics of Stream Processing and Polling Processing.
```

Comment:
Suggested:
An interesting aspect of detection behaviours is the combination of Stream and Polling processing characteristics.